### PR TITLE
Feat/authentication/UI

### PR DESCRIPTION
**Pull request by francelu**
### Overview
This PR introduces the **Sign-In** and **Sign-Up** authentication screens.

### Key Features
- **Sign-In Screen**: 
  - Validates email and password inputs with error messages for invalid entries.
  - Will navigate unregistered users to the **Sign-Up** screen.

- **Sign-Up Screen**: 
  - Validates email, password, and password confirmation with error feedback.

- **AuthSharedComposables.kt**:
  - Shared UI components for consistent design and reduced code duplication.

- **Testing**: 
  - UI tests cover validation and error handling for both screens.

### Next Steps
- Review and provide feedback for merging into the main branch.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -70,9 +70,7 @@ composeViewModel = "2.7.0"
 sonar = "4.4.1.3373"
 
 [libraries]
-androidx-activity-compose = { module = "androidx.activity:activity-compose", version.ref = "activityCompose" }
```
Comment (charliemangano) [inline gradle/libs.versions.toml:73]: This should stay

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -70,9 +70,7 @@ composeViewModel = "2.7.0"
...
 androidx-appcompat = { module = "androidx.appcompat:appcompat", version.ref = "appcompat" }
-androidx-compose-bom = { module = "androidx.compose:compose-bom", version.ref = "composeBom" }
```
Comment (charliemangano) [inline gradle/libs.versions.toml:75]: This should stay

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -81,16 +79,15 @@ androidx-fragment-ktx = { module = "androidx.fragment:fragment-ktx", version.ref
...
 androidx-material = { module = "androidx.compose.material:material", version.ref = "material" }
+androidx-material-icons-extended = { module = "androidx.compose.material:material-icons-extended" }
 androidx-material3 = { module = "androidx.compose.material3:material3", version.ref = "material3" }
 androidx-navigation-compose = { module = "androidx.navigation:navigation-compose", version.ref = "navigationCompose" }
 androidx-navigation-fragment-ktx = { module = "androidx.navigation:navigation-fragment-ktx", version.ref = "navigationCompose" }
 androidx-navigation-ui-ktx = { module = "androidx.navigation:navigation-ui-ktx", version.ref = "navigationCompose" }
 androidx-runner = { module = "androidx.test:runner", version.ref = "runner" }
 androidx-ui = { module = "androidx.compose.ui:ui", version.ref = "ui" }
 androidx-ui-graphics = { module = "androidx.compose.ui:ui-graphics" }
-androidx-ui-test-junit4 = { module = "androidx.compose.ui:ui-test-junit4", version.ref = "uiTestJunit4" }
```
Comment (charliemangano) [inline gradle/libs.versions.toml:91]: This should stay for compatibility with other branches

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -81,16 +79,15 @@ androidx-fragment-ktx = { module = "androidx.fragment:fragment-ktx", version.ref
...
 androidx-ui-test-junit4-v105 = { module = "androidx.compose.ui:ui-test-junit4", version.ref = "uiTestJunit4Version" }
-androidx-ui-test-manifest = { module = "androidx.compose.ui:ui-test-manifest", version.ref = "uiTestManifest" }
```
Comment (charliemangano) [inline gradle/libs.versions.toml:93]: This should stay for compatibility with other branches

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -103,9 +100,7 @@ json = { module = "org.json:json", version.ref = "json" }
 junit = { module = "junit:junit", version.ref = "junit" }
 kaspresso = { module = "com.kaspersky.android-components:kaspresso", version.ref = "kaspresso" }
 kaspresso-allure-support = { module = "com.kaspersky.android-components:kaspresso-allure-support", version.ref = "kaspressoAllureSupport" }
-kaspresso-compose-support = { module = "com.kaspersky.android-components:kaspresso-compose-support", version.ref = "kaspressoComposeSupport" }
```
Comment (charliemangano) [inline gradle/libs.versions.toml:106]: This should stay for compatibility with other branches

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -103,9 +100,7 @@ json = { module = "org.json:json", version.ref = "json" }
...
 kotlinx-coroutines-test = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-test" }
-kotlinx-serialization-json = { module = "org.jetbrains.kotlinx:kotlinx-serialization-json", version.ref = "kotlinxSerializationJson" }
```
Comment (charliemangano) [inline gradle/libs.versions.toml:108]: This should stay for compatibility with other branches

File: gradle/libs.versions.toml [hunk 7]
```diff
@@ -114,7 +109,6 @@ mockito-android = { module = "org.mockito:mockito-android", version.ref = "mocki
 mockito-core = { module = "org.mockito:mockito-core", version.ref = "mockitoCore" }
 mockito-inline = { module = "org.mockito:mockito-inline", version.ref = "mockitoInline" }
 mockito-kotlin = { module = "org.mockito.kotlin:mockito-kotlin", version.ref = "mockitoKotlin" }
-mockk = { module = "io.mockk:mockk", version.ref = "mockk" }
```
Comment (charliemangano) [inline gradle/libs.versions.toml:117]: This should stay for compatibility with other branches

File: gradle/libs.versions.toml [hunk 8]
```diff
@@ -125,8 +119,6 @@ androidx-junit-ktx = { group = "androidx.test.ext", name = "junit-ktx", version.
 androidx-ui-test-junit4-android = { group = "androidx.compose.ui", name = "ui-test-junit4-android", version.ref = "uiTestJunit4Android" }
 
 auth-kt = { module = "io.github.jan-tennert.supabase:auth-kt" }
-bom = { module = "io.github.jan-tennert.supabase:bom", version.ref = "bom" }
```
Comment (charliemangano) [inline gradle/libs.versions.toml:128]: This should stay for compatibility with other branches

File: gradle/libs.versions.toml [hunk 9]
```diff
@@ -125,8 +119,6 @@ androidx-junit-ktx = { group = "androidx.test.ext", name = "junit-ktx", version.
...
-ktor-client-android = { module = "io.ktor:ktor-client-android", version.ref = "ktorClientAndroid" }
```
Comment (charliemangano) [inline gradle/libs.versions.toml:129]: This should stay for compatibility with other branches

**Review by charliemangano — CHANGES_REQUESTED** [review]
Very nice UI! Conforms to the Figma, great job!

### Comments
+ The composable `ErrorMessage` is not directly related to authentication and should be moved to another file. Maybe have `com.android.periodpals.ui.components` folder and put it there. This would improve general organization and make it easier for us later down the line.
+ The `ErrorMessage` composable tests the observable String `message`. I think this should be done outside of it, in `SignInScreen` and `SignUpScreen` to improve readability and modularity.

### Improvements
Next time, try and do smaller PRs. The `SignIn.kt` and `SignUp.kt` could have been separated into two smaller PRs. This would make it easier to review as well as improve general team organisation.

### Steps before merge
Everything listed into the "Comments" section.   
Don't forget to pull from main to get the AGP version bug fixed (see #32 #33).

**Review by charliemangano — APPROVED** [review]
lgtm!
Thank you for implementing the changes

**Review by Harrish92 — APPROVED** [review]
**Summary:**

Great work on implementing the sign-in and sign-up screens! The code is well-structured and easy to follow. The use of Compose components and state management is effective and clean.

**Suggestions:**

For future improvements, consider improving the validation logic for an email address. A valid email address typically has four parts:

- Recipient name
- @ symbol
- Domain name
- Top-level domain

This will help in maintaining data integrity. Keep up the good work!

**Pull request merged**